### ISSUE TITLE
Document concurrency model

### ISSUE THREAD

**Issue by amoghe** [go-check/check#62]
From the documentation it is hard to determine whether tests run sequentially or in parallel, and if the setup/teardown for the test happens on the same goroutine as the test itself.

This matters when trying to diagnose failures that have symptoms of tests (and their setup/teardown) clobbering each other. This started happening after moving to go 1.5, and it may be likely that we've written tests badly, but its hard to be sure.

**Comment by kahing** [comment]
I don't believe this framework runs test concurrently for now. #51 tracks that

**Comment by horkhe** [comment]
I noticed that when I run tests as `go test ./...`, tests from different packages overlap, that was proved printing current timestamp at the beginning and at the end of tests. This is indeed a problem for it result in failure of tests that cannot be run in parallel. As a workaround I had to put explicit `go test` command for each package in the project Makefile.

**Comment by pavlo** [comment]
@horkhe,

Under some circumstances, `go test` command itsels runs test in parallel, see this article for more information - https://splice.com/blog/lesser-known-features-go-test:

> Another source of parallelization, albeit not at the granular test function level, but at the package level, is when `go test p1 p2 p3` is called (that is, when it is called with multiple packages to test). In this case, the test binaries of all packages are built then run at the same time.

So, in your case (well, in my case too) the right way to call it would be this:

`go test -p 1 ./...`

Hope this helps!
